ansible/provision-and-execute-tests/Makefile

```diff
@@ -1,18 +1,25 @@
 include ../Ansible.common.mk
 
-.DEFAULT_GOAL := provision-and-execute-tests
+.PHONY: prerelease
+prerelease: dependencies prepare-inventory
+	ansible-playbook $(ANSIBLE_FOLDER)/playbook-provision-prerelease.yml -i $(ANSIBLE_INVENTORY)
+	ansible-playbook $(ANSIBLE_FOLDER)/playbook-run-tests.yml -i $(ANSIBLE_INVENTORY)
+	ansible-playbook $(ANSIBLE_FOLDER)/playbook-windows.yml -i $(ANSIBLE_INVENTORY)
+	ansible-playbook $(ANSIBLE_FOLDER)/playbook-merge-results.yml -i $(ANSIBLE_INVENTORY)
 
-.PHONY: provision-and-execute-tests
-provision-and-execute-tests: ansible/dependencies ansible/prepare-inventory
-ifeq ($(NRIA_ENV),prerelease)
-	@echo "Running commands for prerelease"
-	ansible-playbook $(ANSIBLE_FOLDER)/playbook-prerelease.yml -i $(ANSIBLE_INVENTORY)
-	TEST_REPORT_FILENAME="test-report-$(NRIA_ENV).xml" ansible-playbook $(ANSIBLE_FOLDER)/playbook-localhost.yml -i $(ANSIBLE_INVENTORY)
-else ifneq (,$(findstring $(NRIA_ENV),staging production))
-	@echo "Running commands for either staging or production"
-	ansible-playbook $(ANSIBLE_FOLDER)/playbook-nr-repo.yml -i $(ANSIBLE_INVENTORY)
-	TEST_REPORT_FILENAME="test-report-$(NRIA_ENV).xml" ansible-playbook $(ANSIBLE_FOLDER)/playbook-localhost.yml -i $(ANSIBLE_INVENTORY)
-endif
+.PHONY: staging
+staging: dependencies prepare-inventory
+	ansible-playbook $(ANSIBLE_FOLDER)/playbook-provision-staging.yml -i $(ANSIBLE_INVENTORY)
+	ansible-playbook $(ANSIBLE_FOLDER)/playbook-run-tests.yml -i $(ANSIBLE_INVENTORY)
+	ansible-playbook $(ANSIBLE_FOLDER)/playbook-merge-results.yml -i $(ANSIBLE_INVENTORY)
 
+.PHONY: production
+production: dependencies prepare-inventory
+	ansible-playbook $(ANSIBLE_FOLDER)/playbook-provision-production.yml -i $(ANSIBLE_INVENTORY)
+	ansible-playbook $(ANSIBLE_FOLDER)/playbook-run-tests.yml -i $(ANSIBLE_INVENTORY)
+	ansible-playbook $(ANSIBLE_FOLDER)/playbook-merge-results.yml -i $(ANSIBLE_INVENTORY)
```

Comment:
⚠️ It seems we're missing to execute `playbook-windows.yml` in these targets.

Reply:
Windwos is a different use case as it's bundled with NRIA. We can just test the package. Once NRIA releases and bundles a new version, tests should be done on their side.